applicationset/services/scm_provider/github.go

```diff
@@ -120,7 +120,7 @@ func (g *GithubProvider) RepoHasPath(ctx context.Context, repo *Repository, path
 func (g *GithubProvider) listBranches(ctx context.Context, repo *Repository) ([]github.Branch, error) {
 	// If we don't specifically want to query for all branches, just use the default branch and call it a day.
 	if !g.allBranches {
-		defaultBranch, _, err := g.client.Repositories.GetBranch(ctx, repo.Organization, repo.Repository, repo.Branch)
+		defaultBranch, _, err := g.client.Repositories.GetBranch(ctx, repo.Organization, repo.Repository, repo.Branch, 0)
```

Comment:
`0` is in the `maxRedirects` param slot.

Following redirects were once [not supported](https://github.com/google/go-github/pull/1901/files).

`0` [means](https://github.com/google/go-github/pull/2939/files) "do not follow redirects" and matches the current behavior.